src/main.ts

```diff
@@ -16,48 +21,101 @@ async function commandExists(cmd: string): Promise<boolean> {
   return code === 0
 }
 
+async function runSbt(baseDir: string, pluginVersionInput: string): Promise<void> {
+  const projectDir = path.join(baseDir, 'project')
+
+  if (!fs.existsSync(projectDir)) {
+    core.setFailed(`${baseDir} is not a valid sbt project: missing folder '${projectDir}'.`)
+    return
+  }
+
+  const uuid = crypto.randomUUID()
+  const pluginFile = path.join(projectDir, `github-dependency-graph-${uuid}.sbt`)
+
+  const pluginVersion =
+    pluginVersionInput.length === 0 ? defaultSbtPluginVersion : pluginVersionInput
+
+  const pluginDep = `addSbtPlugin("ch.epfl.scala" % "sbt-github-dependency-graph" % "${pluginVersion}")`
+  await fsPromises.writeFile(pluginFile, pluginDep)
+
+  const sbtExists = await commandExists('sbt')
+  if (!sbtExists) {
+    core.setFailed('Not found sbt command')
+    return
+  }
+
+  const input = {
+    projects: core
+      .getInput('projects')
+      .split(' ')
+      .filter(value => value.length > 0),
+    scalaVersions: core
+      .getInput('scala-versions')
+      .split(' ')
+      .filter(value => value.length > 0),
+  }
+
+  await cli.exec('sbt', [`githubSubmitDependencyGraph ${JSON.stringify(input)}`], {
+    cwd: baseDir,
+  })
+}
+
+function getMillPath(baseDir: string): string {
```

Comment:
My approach here is just to check if there is a `mill` or `millw` in the workspace and if so, use it. If not, we just grab the official launcher script. It stinks a bit since the name changes every release so we can't target `..../releases/latest/mill` for example because it won't be named `mill`, so that's the reason for the specific tag and version.

Reply:
It also might look a little weird because I'm returning like `./mill` but since we do a cwd when we execute the command I want to pass it the actual file path which `exec` seems to support. I also want to ensure that we do run it from that location because it will then pick up the `.mill-version` if one exists.

Reply:
You could return the `millPath` rather than `./mill` or `millwPath` rather than `./millw` and it would work the same. But I am fine with it.

Reply:
If the `baseDir` is `./foo` and then `mill` file ends up being `./foo/mill` won't this screw it up if we `cwd` into `foo` making that the current working dir if the `millPath` we returned was `.foo/mill` since it'd no longer be correct?